Net011/TrainingEntity.cs

```diff
@@ -5,10 +5,10 @@ namespace Net011
     public abstract class TrainingEntity
     {
         private string _description;
-        private Guid _id;
         public Guid Id
         {
-            get { return _id; }
+            get;
+            set;
         }
         public string Description
         {
```

Comment:
1. Change the order of checks, if value is null you will have exception in line 21
2. Use const int value instead of 256, magic number